### PR TITLE
[lab-supervised-learning] rute correia

### PR DESCRIPTION
**Pull request by aariops**
started

### CODE REVIEW

**Comment by aariops** [comment]
I feel like a code monkey :see_no_evil: 

Anyway, this is done

**Comment by ta-data-lis** [comment]
Lab checked, good job!

Hey monkey you did a great job already... anyway here are is a review to make you feel more human. 

**Challenge 2**
Collinearity, aka columns with high correlation between them, should be dropped because they represent redundant info which you don't want when you are fitting your model. In order to decide which columns are "too correlated to be accepted" usually we decide a threshold upfront (which depends on field, dataset, goal etc...), in this case it was 90% so you should drop ALL the columns with correlation in the order of +90% or -90%.

So also one between: **NUMBER_SPECIAL_CHARACTERS** and **URL_LENGTH**
And, probably also one between: **APP_PACKETS** and **SOURCE_APP_BYTES**

**CHALLENGE 5**
Looking at the result of **get_dummies** on COUNTRY, it seems that the way you substituted all the Countries below the top 10 in CHALLENGE 4 did not work... there is no column "OTHER" ;).
Maybe like this:
**websites['WHOIS_COUNTRY'] = np.where(websites['WHOIS_COUNTRY'].isin(whois['index']), websites['WHOIS_COUNTRY'], 'OTHER')**

**CHALLENGE 6**
_
/home/rute/anaconda3/lib/python3.7/site-packages/sklearn/linear_model/_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)_

This warning means that, as says, it should be specified the "solver" of the model, which is actually the mathematical tool through which the model establishes the best fit iteration after iteration. Depending on the structure of the solver and of the data that we want to model, a different number of iterations is needed... when they are not enough to reach an OPTIMAL mathematical solution (min of cost function) then Python complains a bit... like in this case!

If you use the accuracy method you imported to compare the results of .predict BEFORE VS AFTER **specifying the solver AND increasing the NUMBER OF ITERATIONS** (all in sklearn documentation for ANY MODEL)... you will see that:

Warning message disapperas
The performance of the model improves (cause the "solver" has more room to reach a better solution)

It feels all a little overwhelming, we know, we have been there... the logic is the most important thing, feel free to reach out anytime in case.

Keep up the good work!
